.github/workflows/rust.yml

```diff
@@ -105,6 +105,7 @@ jobs:
           --verbose
           --workspace
           --exclude compile-errors
+          --exclude dusty
```

Comment:
Clap requires `1.70.0` essentially which means dusty won't work in these older nightly version tests. I figure since it's mostly auxiliary to the other tooling here then that's likely fine. If that isn't the case, let me know and I can revert the switch to clap and other related changes.

Reply:
We can bump these toolchains. They're pinned so that we have consistent output for the tests that verify error messages and other compiler outputs, but bumping them and updating the tests is perfectly fine.

Reply:
Could we defer this to another PR?